app/Models/Signup.php

```diff
@@ -65,21 +65,66 @@ public function posts()
     }
 
     /**
-     * Get the visible posts associated with this signup.
+     * Get the visible posts associated with this signup, optionally by type.
      *
+     * @param string Optional param to specify post type.
      * @return \Illuminate\Database\Eloquent\Relations\HasMany
      */
-    public function visiblePosts()
+    public function visiblePosts($type = null)
     {
         if (! is_staff_user()) {
-            return $this->hasMany(Post::class)->where('status', '=', 'accepted')
+            if ($type === null) {
+                return $this->hasMany(Post::class)->where('status', '=', 'accepted')
+                                                  ->orWhere('northstar_id', auth()->id())
+                                                  ->with('tags');
+            }
+
+            return $this->hasMany(Post::class)->whereIn('type', array_values($type))
+                                              ->where('status', '=', 'accepted')
                                               ->orWhere('northstar_id', auth()->id())
-                                              ->with('tags');
+                                              ->with('tags')
+                                              ->get();
```

Comment:
Everything is working except when using include params as an owner (e.g. api/v3/signups?include=posts:type(text|photo) still returns voter-reg posts. I tried to do what is commented out below but none returned the correct response either. I'm wondering if there is a certain order or way I'd have to use whereIn, where, and orWhere when using them all together? @DoSomething/team-bleed